### ISSUE TITLE
TS Error: 'QueryMethod' refers to a value, but is being used as a type here.

### ISSUE THREAD

**Issue by richardrobinson0924** [typegoose/typegoose#465]
## Versions

- System: macOS
- NodeJS: 14.10.1
- Typescript: 4.0.3
- Compiler / Transpiler: tsc
- Typegoose(NPM): 7.4.5
- mongoose: 5.11.8

## What is the Problem?

When creating a `QueryHelpers` interface, I get a TypeScript error when I define the properties as `QueryMethod`:

```
'QueryMethod' refers to a value, but is being used as a type here. Did you mean 'typeof QueryMethod'?
```

## Code Example

```ts
// an interface is needed to add query-function types to the class
interface QueryHelpers {
  // use the actual function types dynamically
  findByName: QueryMethod<typeof findByName>;
  findByLastname: QueryMethod<typeof findByLastname>;
}

function findByName(this: ReturnModelType<typeof Person, QueryHelpers>, name: string) {
  return this.find({ name }); // important to not do an "await" and ".exec"
}
@queryMethod(findByName)
class Person {
  @prop({ required: true })
  public name: string;
}
const PersonModel = getModelForClass(Person);
```

## Do you know *why* it happens?

*no*

**Comment by hasezoey** [comment]
why create one issue and then close it? (#464)

**Comment by hasezoey** [comment]
aside from that, highest supported mongoose version is `mongoose@5.10.18`, and requires `@types/mongoose`

**Comment by richardrobinson0924** [comment]
> why create one issue and then close it? (#464)

@hasezoey Apologies; it was because when I first created the issue, Github redirected me to a 404 page so I assumed it wasn't created, so I created another one. Then I realized both were created, so I closed the first one.

**Comment by malins** [comment]
Hello!

I have exactly the same problem, and I'm using mongoose 5.10.17 (with @types/mongoose).

**Comment by hasezoey** [comment]
@malins could you post your used code, or is it the "exact" same?

**Comment by hasezoey** [comment]
found the problem, thanks for finding this

issue was the decorator was not just exported as `queryMethod` (camelCase), but also as `QueryMethod` (PascalCase), which is the same name as the type `QueryMethod`

**Comment by hasezoey** [comment]
fixed with 5e771373520a41c5568119f41c9afb425c1338ce

released in 7.4.7
